test/compress/unicode.js

```diff
@@ -35,9 +36,43 @@ unicode_identifier_ascii_only: {
 }
 
 unicode_string_literals: {
-    beautify = {ascii_only: true}
+    beautify = {ascii_only: true, ecma: 6}
     input: {
         var a = "6 length unicode character: \u{101111}";
     }
     expect_exact: 'var a="6 length unicode character: \\u{101111}";'
 }
+
+unicode_output_es5_surrogates: {
+    beautify = {ascii_only: true, ecma: 5}
+    input: {
+        var \u{10000} = "6 length unicode character: \u{10FFFF}";
+    }
+    expect_exact: 'var \\ud800\\udc00="6 length unicode character: \\udbff\\udfff";'
+}
+
+check_escape_style: {
+    beautify = {ascii_only: true, ecma: 6}
+    input: {
+        var a = "\x01";
+        var \ua0081 = "\x10"; // \u0081 only in ID_Continue
+        var \u0100 = "\u0100";
+        var \u1000 = "\u1000";
+        var \u{10000} = "\u{10000}";
+        var \u{2f800} = "\u{100000}";
+    }
+    expect_exact: 'var a="\\x01";var \\ua0081="\\x10";var \\u0100="\\u0100";var \\u1000="\\u1000";var \\u{10000}="\\u{10000}";var \\u{2f800}="\\u{100000}";'
+}
+
+check_escape_style_es5: {
+    beautify = {ascii_only: true, ecma: 5}
+    input: {
+        var a = "\x01";
+        var \ua0081 = "\x10"; // \u0081 only in ID_Continue
+        var \u0100 = "\u0100";
+        var \u1000 = "\u1000";
+        var \u{10000} = "\u{10000}";
+        var \u{2f800} = "\u{100000}";
```

Comment:
This seems to go at the limit of the parser in my atom editor and github. But should be valid: http://www.fileformat.info/info/unicode/char/2f800/index.htm